docs/docs/domain.mdx

```diff
@@ -465,64 +470,180 @@ stories:
+- entity_name
+slots:
+  slot_name:
+    type: any
+    mappings:
+    - type: from_entity
+      entity: entity_name
+      role: role_name
+      group: group name
+      intent: intent_name
+      not_intent: excluded_intent
 ```
 
-In this case, you don't have to include the `slot_was_set` part in the
-story, because it is automatically picked up:
+:::note
+Within the context of a form, a slot with `from_entity` mapping will be set only if the mapping
+is unique, unless the slot is requested. You can find out more in [Forms](./forms.mdx#unique-entity-mapping-matching).
+:::
+
+#### from_text
+
+The `from_text` mapping will use the text of the next user utterance to fill the slot
+`slot_name`. If `intent_name` is `None`, the slot will be filled regardless of intent name.
+Otherwise, the slot will only be filled if the user's intent is `intent_name`.
+
+The slot mapping will not apply if the intent of the message is `excluded_intent`.
+Note that you can define lists of intents for the parameters `intent` and `not_intent`.
 
 ```yaml-rasa
-stories:
-- story: entity slot-filling
-  steps:
-  - intent: greet
-    entities:
-    - name: Ali
-  - action: utter_greet_by_name
+slots:
+  slot_name:
+    type: text
+    mappings:
+    - type: from_text
+      intent: intent_name
+      not_intent: excluded_intent
 ```
 
+#### from_intent
 
-:::note Auto-filled slots & `influence_conversation`
-An auto-filled slot that is defined with `influence_conversation: true` will influence the conversation the same way any other slot does.
+The `from_intent` mapping will fill slot `slot_name` with value `my_value` if
+user intent is `intent_name` or `None`. The slot mapping will not
+apply if the intent of the message is `excluded_intent`. Note that you can
+also define lists of intents for the parameters `intent` and `not_intent`.
 
-In the example above, if the `name` slot's type is `text`, then it only matters that some name was detected, but it won't matter which one.
-If the `name` slot is type `categorical`, then the behaviour will vary depending on the categories you have defined for the slot. 
+```yaml-rasa
+slots:
+  slot_name:
+    type: any
+    mappings:
+    - type: from_intent
+      value: my_value
+      intent: intent_name
+      not_intent: excluded_intent
+```
 
-Explicitly including a `slot_was_set` step in your story can make the behaviour of an
-auto-filled slot that influences the conversation clearer, and will not change the behaviour of your story.
+#### from_trigger_intent
 
-:::
+The `from_trigger_intent` mapping will fill slot `slot_name` with value `my_value`
+if a form is activated by a user message with intent `intent_name`.
+The slot mapping will not apply if the intent of the message is
+`excluded_intent`. Note that you can
+also define lists of intents for the parameters `intent` and `not_intent`.
+
+A slot with `from_trigger_intent` mapping must always include mapping conditions
+and be listed in the form's `required_slots`.
```

Comment:
I'm actually going to change my mind about the first half of the sentence after rechecking the code, doesn't need to include mapping conditions, only needs to be in the form's required slots. This can also be checked with `rasa data validate`, should I link to that info instead?

Reply:
Yes, that makes sense!